tests/unit/apps/test_command_line_utilities.py

```diff
@@ -71,6 +71,34 @@ def test_get_instruments_data(self, _, args_list):
             self.assertEqual(data.tail(1).values[0][0], "Whitebread_LondonStockEx_WTB")
         else:
             self.assertEqual(data.tail(1).values[0][0], "USTreasury_6.875_2025")
+
+    @parameterized.expand(
+        [
+            [
+                "test_xlsx_file",
+                {
+                    "file_path": os.path.join(
+                        cur_dir, test_data_root.joinpath("instruments.xlsx")
+                    ),
+                    "num_header": 0,
+                    "num_footer": 0,
+                    "delimiter": None,
+                    "scale_quotes": False,
+                    "line_terminator": None,
+                },
+            ]
+        ]
+    )
+    def test_get_instruments_data_excel(self, _, args_list):
+        data = load_data_to_df_and_detect_delimiter(args_list)
+        if args_list["num_header"] == 0:
```

Comment:
avoid having conditional logic in test asserts, suggests they should be separate tests